lib/takeaway.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'menu'
+require_relative 'textsender'
+
+class Takeaway 
+  attr_reader :menu, :basket
+
+  def initialize(menu = Menu.new)
+    @menu = menu
+    @basket = Hash.new(0)
+    @sum = 0
+    
+  end
+
+  def view_menu
+    @menu.display
+  end
+
+  def select_dish(dish, quantity)
+    @basket[dish] += quantity
+  end
+
+  def total
+    @sum = @basket.reduce(0) do |sum, (dish, quantity)| 
+    sum + (@menu.menu[dish] * quantity)
+    end
+    p "Total is £#{@sum}"
+  end
+
+  def view_basket
+    @basket.each do |dish, quantity|
+      p "#{dish}x#{quantity}"
+    end
+    p "Total:£#{@sum}"
+  end
+
+  def place_order(amount)
+    fail "Incorrect total" if amount != @sum
```

Comment:
What happens if you haven't call `#total` yet before you try and place an order?